### ISSUE TITLE
Asymmetric property with deleter shouldn't narrow type on assignment

### ISSUE THREAD

**Issue by shiftinv** [microsoft/pyright#3547]
**Describe the bug**
An asymmetric property with a `deleter` does not skip type narrowing on assignment (#2626) - without a deleter, type narrowing doesn't take effect, as expected.

**To Reproduce**
See code below.
By removing the `@prop.deleter` or putting it above the setter, the second `assert_type` passes fine.

**Expected behavior**
The inferred type of an asymmetric property access should not change after assignment to the property.

**Screenshots or Code**
```python
from typing_extensions import assert_type

class A:
    @property
    def prop(self) -> str:
        ...

    @prop.setter
    def prop(self, value: int):
        ...

    @prop.deleter
    def prop(self):
        ...

a = A()

assert_type(a.prop, str)
a.prop = 42
assert_type(a.prop, str)  # "assert_type" mismatch: expected "str" but received "Literal[42]"
```

**VS Code extension or command-line**
cli, 1.1.252

**Additional context**
I'm not at all familiar with the code, but from a bit of debugging it appears that `isAsymmetricDescriptor` is never propagated to the class type generated for the deleter, meaning this expression will always be false:
https://github.com/microsoft/pyright/blob/22dbd0eee392e4f147d11c385a791ee4aead3028/packages/pyright-internal/src/analyzer/properties.ts#L243

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by shiftinv** [comment]
Thanks, much appreciated!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.253, which I just published. It will also be included in the next release of pylance.
